lib/src/compile.ts

```diff
@@ -0,0 +1,163 @@
+// Copyright 2020 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import {RawSourceMap} from 'source-map';
+import {URL} from 'url';
+
+import {EmbeddedCompiler} from './embedded/compiler';
+import {Dispatcher} from './embedded/dispatcher';
+import {MessageTransformer} from './embedded/message-transformer';
+import {PacketTransformer} from './embedded/packet-transformer';
+import {SassException} from './exception/exception';
+import {deprotifySourceSpan} from './utils';
+import {InboundMessage} from './vendor/embedded_sass_pb';
+
+/**
+ * Compiles a path and returns the resulting css. Throws a SassException if the
+ * compilation failed.
+ */
+export async function compile(options: {
+  path: string;
+  sourceMap?: (sourceMap: RawSourceMap) => void;
+}): Promise<string> {
+  // TODO(awjin): Create logger, importer, function registries.
+
+  const request = newCompileRequest({
+    path: options.path,
+    sourceMap: !!options.sourceMap,
+  });
+
+  const response = await compileRequest(request);
+  if (options.sourceMap && response.sourceMap) {
+    options.sourceMap(response.sourceMap);
+  }
+  return response.css;
+}
+
+/**
+ * Compiles a string and returns the resulting css. Throws a SassException if
+ * the compilation failed.
+ */
+export async function compileString(options: {
+  source: string;
+  sourceMap?: (sourceMap: RawSourceMap) => void;
+  url?: URL;
+}): Promise<string> {
+  // TODO(awjin): Create logger, importer, function registries.
+
+  const request = newCompileStringRequest({
+    source: options.source,
+    sourceMap: !!options.sourceMap,
+    url: options.url,
+  });
+
+  const response = await compileRequest(request);
+  if (options.sourceMap && response.sourceMap) {
```

Comment:
Shouldn't it be an error if `options.sourceMap` is set and `response.sourceMap` is empty?

Reply:
Yes, good catch